include/mbedtls/ssl.h

```diff
@@ -740,43 +740,64 @@ mbedtls_ssl_states;
 
 #if defined(MBEDTLS_SSL_EARLY_DATA) && defined(MBEDTLS_SSL_CLI_C)
 typedef enum {
-/*
- * The client has not sent the first ClientHello yet, it is unknown if the
- * client will send an early data indication extension or not.
- */
-    MBEDTLS_SSL_EARLY_DATA_STATUS_UNKNOWN,
-
 /*
  * See documentation of mbedtls_ssl_get_early_data_status().
  */
-    MBEDTLS_SSL_EARLY_DATA_STATUS_NOT_SENT,
+    MBEDTLS_SSL_EARLY_DATA_STATUS_NO_IND_SENT,
     MBEDTLS_SSL_EARLY_DATA_STATUS_ACCEPTED,
     MBEDTLS_SSL_EARLY_DATA_STATUS_REJECTED,
+} mbedtls_ssl_early_data_status;
+
+typedef enum {
```

Comment:
Ah, no, wait: since this is about the internal state, should we really define it in public? Shouldn't we move that to `ssl_misc.h`?

Reply:
The enum type is used in the SSL context for the `early_data_state` thus we cannot move it to ssl_misc.h. I would prefer to move the definition of MBEDTLS_SSL_EARLY_DATA_STATE_ to ssl_misc.h but this cannot be an enum in that case and the type of `early_data_state` would be `int`. @mpg @tom-cosgrove-arm would you be ok with this?

Reply:
> The enum type is used in the SSL context for the `early_data_state` thus we cannot move it to ssl_misc.h.

Ah, good point, missed that.

> I would prefer to move the definition of MBEDTLS_SSL_EARLY_DATA_STATE_ to ssl_misc.h but this cannot be an enum in that case and the type of `early_data_state` would be `int`

Sounds good. C's type system doesn't really distinguish between `enum` types and other integral types anyway. (I mean, even in a variable has type `enum foo` you can still assign it `2` and the compiler won't complain. So, it's not like we're loosing any type checks by moving to an `int`)

Reply:
> if a variable has type `enum foo` you can still assign it `2` and the compiler won't complain

That is, if `2` is a legal value for `enum foo`. If it's out of range, then the versions of GCC and Clang on my machine still won't complain with "only" `-Wall -Wextra` but clang will complain with `-Weverything` (specifically, `-Wassign-enum`). 

Even with `-Weverything`, if 2 is a legal value for this enum, clang will not complain about assigning `2` (as opposed to whatever enum constant has this value).

Reply:
> this cannot be an enum in that case and the type of `early_data_state` would be `int`

Yes, that's fine

Reply:
To me all together it is better for MBEDTLS_SSL_EARLY_DATA_STATE_  to be declared in ssl_misc.h not in an enum as opposed to be declared to ssl.h (public header) as part of an enum. @tom-cosgrove-arm would you be okay with that?

Reply:
> > this cannot be an enum in that case and the type of `early_data_state` would be `int`
> 
> Yes, that's fine

okay I am doing so then

Reply:
I've eventually kept the mbedtls_ssl_early_data_state enum but moved its definition to ssl_misc.h and adapted the SSL context accordingly.